plan/physical_plans.go

```diff
@@ -561,6 +596,7 @@ func buildJoinSchema(joinType JoinType, join Plan, outerID int) *expression.Sche
 		newSchema.Append(join.Schema().Columns[join.Schema().Len()-1])
 		return newSchema
 	}
+	resetDefaultValues(joinType, join)
```

Comment:
how about `resetDefaultValues(joinType, join, outerID)`, then in function `resetDefaultValues` we can get `childSchemaLen` directly by `join.Children()[outerID].Schema().Len()`

Reply:
maybe no difference, the function still has to check LeftOuterJoin and RightOuterJoin @zz-jason